src/main/scala/mesosphere/marathon/state/GroupManagement.scala

```diff
@@ -0,0 +1,79 @@
+package mesosphere.marathon.state
+
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.upgrade.DeploymentPlan
+
+import scala.concurrent.Future
+
+trait GroupManagement {
```

Comment:
:+1: for making a trait for this. But I don't like the naming here too much: `GroupManagement` vs `GroupManager`.

[If you rename, do it in a separate request, though]